runtime/lua/vim/provider/health.lua

```diff
@@ -4,96 +4,49 @@ local iswin = vim.fn.has('win32') == 1
 local M = {}
 
 local function cmd_ok(cmd)
-  local out = vim.fn.system(cmd)
-  return vim.v.shell_error == 0, out
+  local result = vim.system(cmd, { text = true }):wait()
+  return result.code == 0, result.stdout
 end
 
--- Attempts to construct a shell command from an args list.
--- Only for display, to help users debug a failed command.
---- @param cmd string|string[]
-local function shellify(cmd)
-  if type(cmd) ~= 'table' then
-    return cmd
```

Comment:
why remove this? it was designed to handle `string[]` list commands. it does more than just table.concat()